.github/workflows/build.yml

```diff
@@ -8,12 +8,44 @@ on:
     branches:
       - master
 
+env:
+  PHP_EXTENSIONS: apcu, memcached, redis
+
 jobs:
-  hello:
-    name: Hello
+  static-analysis:
+    name: Static Analysis
 
     runs-on: ubuntu-latest
 
+    strategy:
+      matrix:
+        php-version:
+          - 7.2
+
     steps:
-      - name: Echo a greeting
-        run: echo 'Hello, GitHub Actions!'
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Install PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          coverage: none
+          extensions: ${{ env.PHP_EXTENSIONS }}
```

Comment:
Referencing the environment variable `PHP_EXTENSIONS` here so we can configure extensions we need.

💁‍♂️ For reference, see:

- https://github.com/shivammathur/setup-php#heavy_plus_sign-php-extension-support
- https://docs.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables

Reply:
I think `tests` will be the only job to require PHP extensions to be installed.

Reply:
Not sure, perhaps installation of dependencies with `composer` will fail otherwise?

We can try.

Reply:
Looks like it works as expected!

Reply:
> perhaps installation of dependencies with composer will fail otherwise?

@localheinz That was actually really good point, I'm surprised it works. 🤔 https://github.com/nikolaposa/rate-limit/blob/master/composer.json#L25

Reply:
Let's dump the installed extensions quickly.

Reply:
See https://pipelines.actions.githubusercontent.com/rIXpeNMczGKzgSGv0cMqGFc6K0EVAgSCwhkn3poOmYagKCWORZ/_apis/pipelines/1/runs/5/signedlogcontent/3?urlExpires=2020-08-10T11%3A07%3A42.3116473Z&urlSigningMethod=HMACV1&urlSignature=PKR2fedgBpDcC3KZV6kHZkQsJM19LfYHXfrAGhwk3Uk%3D:

```
2020-08-10T11:06:01.3625168Z [PHP Modules]
2020-08-10T11:06:01.3625338Z amqp
2020-08-10T11:06:01.3625459Z apcu
2020-08-10T11:06:01.3625576Z bcmath
2020-08-10T11:06:01.3625677Z bz2
2020-08-10T11:06:01.3625791Z calendar
2020-08-10T11:06:01.3625914Z Core
2020-08-10T11:06:01.3626027Z ctype
2020-08-10T11:06:01.3626139Z curl
2020-08-10T11:06:01.3626250Z date
2020-08-10T11:06:01.3626347Z dba
2020-08-10T11:06:01.3626458Z dom
2020-08-10T11:06:01.3626572Z enchant
2020-08-10T11:06:01.3626686Z exif
2020-08-10T11:06:01.3626823Z fileinfo
2020-08-10T11:06:01.3626936Z filter
2020-08-10T11:06:01.3627048Z ftp
2020-08-10T11:06:01.3627145Z gd
2020-08-10T11:06:01.3627259Z gettext
2020-08-10T11:06:01.3627370Z gmp
2020-08-10T11:06:01.3627480Z hash
2020-08-10T11:06:01.3627593Z iconv
2020-08-10T11:06:01.3627704Z igbinary
2020-08-10T11:06:01.3627830Z imap
2020-08-10T11:06:01.3627943Z interbase
2020-08-10T11:06:01.3628056Z intl
2020-08-10T11:06:01.3628153Z json
2020-08-10T11:06:01.3628261Z ldap
2020-08-10T11:06:01.3628370Z libxml
2020-08-10T11:06:01.3628482Z mbstring
2020-08-10T11:06:01.3628596Z memcache
2020-08-10T11:06:01.3628709Z memcached
2020-08-10T11:06:01.3628809Z mongodb
2020-08-10T11:06:01.3628920Z msgpack
2020-08-10T11:06:01.3629032Z mysqli
2020-08-10T11:06:01.3629143Z mysqlnd
2020-08-10T11:06:01.3629253Z odbc
2020-08-10T11:06:01.3629363Z openssl
2020-08-10T11:06:01.3629472Z pcntl
2020-08-10T11:06:01.3629570Z pcre
2020-08-10T11:06:01.3629680Z PDO
2020-08-10T11:06:01.3629791Z pdo_dblib
2020-08-10T11:06:01.3629906Z PDO_Firebird
2020-08-10T11:06:01.3630025Z pdo_mysql
2020-08-10T11:06:01.3630138Z PDO_ODBC
2020-08-10T11:06:01.3630237Z pdo_pgsql
2020-08-10T11:06:01.3630349Z pdo_sqlite
2020-08-10T11:06:01.3630461Z pgsql
2020-08-10T11:06:01.3630572Z Phar
2020-08-10T11:06:01.3630682Z posix
2020-08-10T11:06:01.3630791Z pspell
2020-08-10T11:06:01.3630905Z readline
2020-08-10T11:06:01.3631005Z recode
2020-08-10T11:06:01.3631114Z redis
2020-08-10T11:06:01.3631226Z Reflection
2020-08-10T11:06:01.3631337Z session
2020-08-10T11:06:01.3631447Z shmop
2020-08-10T11:06:01.3631557Z SimpleXML
2020-08-10T11:06:01.3631656Z snmp
2020-08-10T11:06:01.3631765Z soap
2020-08-10T11:06:01.3631882Z sockets
2020-08-10T11:06:01.3631994Z sodium
2020-08-10T11:06:01.3632103Z SPL
2020-08-10T11:06:01.3632213Z sqlite3
2020-08-10T11:06:01.3632309Z standard
2020-08-10T11:06:01.3632419Z sysvmsg
2020-08-10T11:06:01.3632529Z sysvsem
2020-08-10T11:06:01.3632638Z sysvshm
2020-08-10T11:06:01.3632747Z tidy
2020-08-10T11:06:01.3632856Z tokenizer
2020-08-10T11:06:01.3632972Z wddx
2020-08-10T11:06:01.3633069Z xml
2020-08-10T11:06:01.3633178Z xmlreader
2020-08-10T11:06:01.3633293Z xmlrpc
2020-08-10T11:06:01.3633403Z xmlwriter
2020-08-10T11:06:01.3633512Z xsl
2020-08-10T11:06:01.3633621Z yaml
2020-08-10T11:06:01.3633718Z Zend OPcache
2020-08-10T11:06:01.3633831Z zip
2020-08-10T11:06:01.3633941Z zlib
2020-08-10T11:06:01.3634054Z zmq
2020-08-10T11:06:01.3634209Z 
2020-08-10T11:06:01.3634310Z [Zend Modules]
2020-08-10T11:06:01.3634415Z Zend OPcache
```

Reply:
I think its best to make the requirements explicit, even if the extensions are installed on the system already, @nikolaposa.

What do you think?

Reply:
> I think its best to make the requirements explicit

Agreed. 👍